bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/firmware/Firmware.java

```diff
@@ -285,11 +299,27 @@ public boolean isPrerequisiteVersion(String firmwareVersion) {
         return new Version(firmwareVersion).compare(internalPrerequisiteVersion) >= 0;
     }
 
+    public boolean isSuitableFor(Thing thing) {
+        return hasSameThingType(thing) && hasRequiredModel(thing);
+    }
+
     @Override
     public int compareTo(Firmware firmware) {
         return -internalVersion.compare(new Version(firmware.getVersion()));
     }
 
+    private boolean hasSameThingType(Thing thing) {
+        return Objects.equal(this.getUID().getThingTypeUID(), thing.getThingTypeUID());
```

Comment:
Note for Java 7 and later: This method should be treated as deprecated; use Objects.equals(java.lang.Object, java.lang.Object) instead. It's now plain Java. Side note: can any of these arguments be null? I doubt, so why not just using: 
```` 
this.getUID().getThingTypeUID().equals(thing.getThingTypeUID())
````

Reply:
Thanks for the hint to the `java.util.Objects`, I will change this.

You are right, the UID of the firmware and the thing type UID in this UID cannot be null. In my opinion, the code using `Objects.equals(...)` is a little easier to read than the `this.getUID().getThingTypeUID().equals(...)` (since then a reader can see right at the beginning that equality is checked, instead of seeing the equals call in the middle of the line). Do you have a reason why this should be changed anyway?